SourceryTests/ParserSpec.swift

```diff
@@ -249,41 +249,58 @@ class ParserSpec: QuickSpec {
                                         Enum(name: "Foo", accessLevel: .internal, isExtension: false, inheritedTypes: [], cases: [], variables: [Variable(name: "x", type: "Int", accessLevel: (.internal, .none), isComputed: true)])
                                 ]))
                     }
+                    
+                    context("given enum without rawType") {
+                        it("extracts inherited types properly") {
+                            expect(parse("enum Foo: SomeProtocol { case optionA }; protocol SomeProtocol {}"))
+                                .to(equal([
+                                    Enum(name: "Foo", accessLevel: .internal, isExtension: false, inheritedTypes: ["SomeProtocol"], rawType: nil, cases: [Enum.Case(name: "optionA")]),
+                                    Protocol(name: "SomeProtocol")
+                                    ]))
+                        }
+
+                        it("extracts types inherited in extension properly") {
+                            expect(parse("enum Foo { case optionA }; extension Foo: SomeProtocol {}; protocol SomeProtocol {}"))
+                                .to(equal([
+                                    Enum(name: "Foo", accessLevel: .internal, isExtension: false, inheritedTypes: ["SomeProtocol"], rawType: nil, cases: [Enum.Case(name: "optionA")]),
+                                    Protocol(name: "SomeProtocol")
+                                    ]))
+                        }
+                    }
 
                     context("given enum containing rawType") {
 
                         it("extracts enums without RawRepresentable") {
-                            let expectedEnum = Enum(name: "Foo", accessLevel: .internal, isExtension: false, inheritedTypes: [], cases: [Enum.Case(name: "optionA")])
-                            expectedEnum.rawType = "String"
-
-                            expect(parse("enum Foo: String { case optionA }")).to(equal([expectedEnum]))
+                            expect(parse("enum Foo: String, SomeProtocol { case optionA }; protocol SomeProtocol {}"))
```

Comment:
nothing to change here but when doing tdd/bdd and refactoring code it's more reliable to do so while going from green to green and changing tests after (in separate commit) because if public interface didn't change then tests aren't in need of changes and it's making reasoning about changes not breaking anything easier.